# Proyecto: Análisis de Desempeño del Equipo
## Versión Estudiante
Complete cada celda siguiendo las instrucciones.

In [2]:
import pandas as pd

## 1. Cargar dataset

In [3]:
df = pd.read_csv('./data/desempeno_equipo_100.csv')
df.head()

,colaborador,area,calificacion_desempeno,meta_trimestral,cumplimiento,edad,salario,anios_en_empresa,kpi_productividad,kpi_calidad
0,Claudia Muñoz,IT,93,95,93,36,809178,8,0.80,0.58
1,Ronald Vargas,Operaciones,65,100,88,24,879365,5,0.98,0.92
2,Josué Soto,Operaciones,75,85,76,43,808795,9,0.65,0.81
3,Jorge Rojas,Operaciones,70,90,67,58,1078178,7,0.59,0.53
4,Sofía León,RRHH,68,88,73,47,618148,2,0.84,0.72


## 2. Exploración inicial

In [4]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 10 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   colaborador             100 non-null    object 
 1   area                    100 non-null    object 
 2   calificacion_desempeno  100 non-null    int64  
 3   meta_trimestral         100 non-null    int64  
 4   cumplimiento            100 non-null    int64  
 5   edad                    100 non-null    int64  
 6   salario                 100 non-null    int64  
 7   anios_en_empresa        100 non-null    int64  
 8   kpi_productividad       100 non-null    float64
 9   kpi_calidad             100 non-null    float64
dtypes: float64(2), int64(6), object(2)
memory usage: 7.9+ KB


In [5]:
df.describe()


,calificacion_desempeno,meta_trimestral,cumplimiento,edad,salario,anios_en_empresa,kpi_productividad,kpi_calidad
count,100.000000,100.000000,100.000000,100.000000,1.000000e+02,100.000000,100.000000,100.000000
mean,75.940000,90.930000,83.190000,42.770000,1.002267e+06,7.280000,0.754300,0.731200
std,13.863563,4.967073,10.462724,10.969061,2.838781e+05,3.934001,0.146186,0.135447
min,55.000000,85.000000,65.000000,22.000000,5.504040e+05,1.000000,0.510000,0.510000
25%,63.000000,88.000000,74.000000,34.750000,7.502050e+05,4.000000,0.620000,0.610000
50%,75.000000,90.000000,83.000000,43.000000,1.012544e+06,7.500000,0.770000,0.710000
75%,88.250000,95.000000,93.000000,52.250000,1.264726e+06,10.000000,0.872500,0.840000
max,99.000000,100.000000,99.000000,59.000000,1.497609e+06,14.000000,0.990000,0.980000


In [6]:
df.isnull().sum()

colaborador               0
area                      0
calificacion_desempeno    0
meta_trimestral           0
cumplimiento              0
edad                      0
salario                   0
anios_en_empresa          0
kpi_productividad         0
kpi_calidad               0
dtype: int64

## 3. Promedios por área

In [7]:
df.groupby('area').mean(numeric_only=True)

,calificacion_desempeno,meta_trimestral,cumplimiento,edad,salario,anios_en_empresa,kpi_productividad,kpi_calidad
area,,,,,,,,
Finanzas,78.500000,91.200000,82.100000,43.900000,1.033835e+06,5.600000,0.770000,0.636000
IT,79.052632,90.473684,85.578947,41.157895,8.982284e+05,8.263158,0.810526,0.754737
Marketing,72.913043,92.652174,84.956522,44.434783,1.002971e+06,7.391304,0.750435,0.718261
Operaciones,81.105263,88.789474,82.631579,42.105263,1.012287e+06,6.842105,0.735263,0.753158
RRHH,72.034483,91.172414,80.965517,42.551724,1.052421e+06,7.413793,0.727586,0.744483


## 4. Top 5 colaboradores

In [9]:
df.nlargest(5, 'calificacion_desempeno')

,colaborador,area,calificacion_desempeno,meta_trimestral,cumplimiento,edad,salario,anios_en_empresa,kpi_productividad,kpi_calidad
45,Daniel Duarte,IT,99,90,93,57,610692,13,0.64,0.59
53,Mario León,IT,99,95,70,52,753196,12,0.78,0.71
54,Ana Pérez,RRHH,99,100,92,41,1261131,8,0.56,0.82
75,Lorena Arce,Finanzas,99,88,99,36,1120559,9,0.87,0.59
81,Sergio Valverde,Operaciones,99,95,69,27,1497609,6,0.86,0.60


## 5. Crear columna de clasificación

In [ ]:
df_modified = df.copy()
df_modified.head()

In [13]:
df_modified['calificacion_desempeno'].describe()

count    100.000000
mean      75.940000
std       13.863563
min       55.000000
25%       63.000000
50%       75.000000
75%       88.250000
max       99.000000
Name: calificacion_desempeno, dtype: float64

### Criterio utilizado para columna 'clasificacion'

Sabemos que la clasificacion de desempeño mínima observada es de 55 y la máxima de 99. Además, los cuartiles son aproximadamente:
- Q1 ≈ **63**
- Q2 (mediana) ≈ **75**
- Q3 ≈ **88.25**

Para simplificar en clase, definiremos los niveles así:

- **Baja**: `calificacion_desempeno < 63`
- **Media**: `4.5 <= calificacion_desempeno <= 88`
- **Alta**: `calificacion_desempeno > 88`

In [16]:
bins = [0, 63, 88, 99]
labels = ["Baja", "Media", "Alta"]

df_modified["clasificacion"] = pd.cut(
    df_modified["calificacion_desempeno"],
    bins=bins,
    labels=labels,
    include_lowest=True
)

df_modified[["colaborador","calificacion_desempeno", "clasificacion"]].head(10)


,colaborador,calificacion_desempeno,clasificacion
0,Claudia Muñoz,93,Alta
1,Ronald Vargas,65,Media
2,Josué Soto,75,Media
3,Jorge Rojas,70,Media
4,Sofía León,68,Media
5,Josué Ríos,62,Baja
6,Valeria Ugalde,72,Media
7,Tatiana Jiménez,94,Alta
8,Óscar Villalobos,69,Media
9,Daniela Cordero,71,Media


## 6. Resumen por categoría

In [17]:
df.groupby('area').describe()

calificacion_desempeno                                          \
                             count       mean        std   min   25%   50%   
area                                                                         
Finanzas                      10.0  78.500000  15.407430  57.0  65.0  79.5   
IT                            19.0  79.052632  12.916285  56.0  69.0  79.0   
Marketing                     23.0  72.913043  13.152646  55.0  60.0  73.0   
Operaciones                   19.0  81.105263  14.177034  57.0  69.0  83.0   
RRHH                          29.0  72.034483  13.436312  55.0  62.0  68.0   

                         meta_trimestral             ... kpi_productividad  \
               75%   max           count       mean  ...               75%   
area                                                 ...                     
Finanzas     91.75  99.0            10.0  91.200000  ...             0.865   
IT           88.50  99.0            19.0  90.473684  ...             0.940   
Marketing    84.50  97.0            23.0  92.652174  ...             0.850   
Operaciones  94.50  99.0            19.0  88.789474  ...             0.850   
RRHH         83.00  99.0            29.0  91.172414  ...             0.840   

                  kpi_calidad                                                  \
              max       count      mean       std   min     25%    50%    75%   
area                                                                            
Finanzas     0.89        10.0  0.636000  0.086436  0.54  0.5825  0.605  0.640   
IT           0.99        19.0  0.754737  0.109563  0.58  0.6750  0.780  0.840   
Marketing    0.99        23.0  0.718261  0.114322  0.54  0.6500  0.690  0.820   
Operaciones  0.98        19.0  0.753158  0.165967  0.51  0.6100  0.750  0.905   
RRHH         0.98        29.0  0.744483  0.150038  0.51  0.6000  0.760  0.850   

                   
              max  
area               
Finanzas     0.80  
IT           0.92  
Marketing    0.93  
Operaciones  0.98  
RRHH         0.98  

[5 rows x 64 columns]

## 7. Exportar archivo final

In [20]:
# Escriba su solución aquí
df_modified.to_csv('./data/desempeno_equipo_100_modified.csv', index=False)